src/main/java/seedu/address/logic/PersonAdapter.java

```diff
@@ -18,29 +29,82 @@ public Person getSubject() {
         return this.subject;
     }
 
-    // TODO: use proper types for field and newValue
-    // Use an enum for field
-    // newValue should be a valid type for the given enum, not sure how, generic?
-
     /**
-     * Accept parameters to edit the Person.
+     * Accept parameters to edit the person
      *
      * @param field to be edited.
-     * @param newValue to replace current value.
-     * @return new Person object.
+     * @param newValue to replace the current value.
+     * @return new Person object which is modified if no errors with the new field input.
      */
-    public Person edit(Object field, Object newValue) {
+    public Person edit(PersonField field, String newValue) {
         Person curr = this.subject;
         Person newPerson = editPerson(curr, field, newValue);
-        model.setPerson(curr, newPerson);
         this.subject = newPerson;
         return newPerson;
     }
 
-    // TODO use field and newValue properly.
-    private Person editPerson(Person personToEdit, Object field, Object newValue) {
-        return new Person(
-                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                personToEdit.getAddress(), personToEdit.getTags());
+    //TODO add edit field for tags.
+    private Person editPerson(Person personToEdit, PersonField field, String newValue) {
+        switch(field) {
+        case NAME:
+            return editPersonName(personToEdit, newValue);
+        case PHONE:
+            return editPersonPhone(personToEdit, newValue);
+        case EMAIL:
+            return editPersonEmail(personToEdit, newValue);
+        case ADDRESS:
+            return editPersonAddress(personToEdit, newValue);
+        default:
+            return personToEdit;
+        }
+    }
+
+    private Person editPersonName(Person personToEdit, String newValue) {
+        try {
+            Name newName = new Name(newValue);
+            return new Person(
+                    newName, personToEdit.getPhone(), personToEdit.getEmail(),
+                    personToEdit.getAddress(), personToEdit.getTags());
+        } catch (Exception e) {
```

Comment:
Would recommend that we not catch exceptions and clear them silently. I believe any exception should be rethrown to be handled in `PersonAdapter.edit(..)`. After that, I suggest `.edit(..)` gives a custom Exception to the component that called it.